src/HaskellCI/GitHub.hs

```diff
@@ -304,7 +324,23 @@ makeGitHub _argv config@Config {..} prj jobs@JobVersions {..} = do
                             change_dir_if vr $ pkgNameDirVariable pkgName
                             sh_if vr $ "doctest " ++ doctestOptions ++ " " ++ args'
 
-        -- TODO: hlint
+        -- hlint
+        when (cfgHLintEnabled cfgHLint) $ githubRun "hlint" $ do
+            let "" <+> ys = ys
+                xs <+> "" = xs
+                xs <+> ys = xs ++ " " ++ ys
+
+                prependSpace "" = ""
+                prependSpace xs = " " ++ xs
+
+            let hlintOptions = prependSpace $ maybe "" ("-h ${TOP}/" ++) (cfgHLintYaml cfgHLint) <+> unwords (cfgHLintOptions cfgHLint)
```

Comment:
there is no `TOP`. It should be `$GITHUB_WORKSPACE/source` (that's where we checkout the packages).